lib/tpm_kdfa.c

```diff
@@ -90,12 +90,26 @@ TPM_RC tpm_kdfa(TPMI_ALG_HASH hashAlg,
         return TPM_RC_HASH;
     }
 
-    HMAC_CTX ctx;
-    HMAC_CTX_init(&ctx);
-    int rc = HMAC_Init_ex(&ctx, key->buffer, key->size, md, NULL);
+    HMAC_CTX *ctx;
+#if OPENSSL_VERSION_NUMBER < 0x1010000fL /* OpenSSL 1.1.0 */
```

Comment:
I was waiting for this to be an issue. Rather than ifdef them inline, can we just create some routines (within tpm_kdfa.c is fine) that handle this for us. This function is already nasty enough and I don't want to make it worse.

maybe something like the following prototypes:

```
HMAC_CTX *hmac_new();
void hmac_free(HMAC_CTX *ctx);
```